sdk/src/main/scala/io/horizen/account/state/McAddrOwnershipMsgProcessor.scala

```diff
@@ -38,24 +40,65 @@ case class McAddrOwnershipMsgProcessor(params: NetworkParams) extends NativeSmar
   override val contractAddress: Address = MC_ADDR_OWNERSHIP_SMART_CONTRACT_ADDRESS
   override val contractCode: Array[Byte] = Keccak256.hash("McAddrOwnershipSmartContractCode")
 
-  // ecdsa curve y^2 mod p = (x^3 + 7) mod p
-  val ecParameters: X9ECParameters = SECNamedCurves.getByName("secp256k1")
+  override def init(view: BaseAccountStateView, consensusEpochNumber: Int): Unit = {
+    if (!isForkActive(consensusEpochNumber)) {
+      log.warn(s"Can not perform ${getClass.getName} initialization, fork is not active")
+      return
+    }
+    else {
```

Comment:
are you sure is needed this else?  I would remove it , should never happen since this init() method can be called:
- at genesys block : the init for sure is not happened befor
- from the canProcess of this class: there is already a if (!initDone(view)) immediately before there

Reply:
I agree, that should not happen. Yet I would prefer having an exception for troubleshooting any possible code change in future.